Plutarch/Internal.hs

```diff
@@ -745,7 +831,25 @@ compile' t =
 -- | Compile a (closed) Plutus Term to a usable script
 compile :: Config -> ClosedTerm a -> Either Text Script
 compile config t = case asClosedRawTerm t of
-  TermMonad (ReaderT t') -> Script . UPLC.Program () uplcVersion . compile' <$> t' config
+  TermMonad (ReaderT t') -> do
+    configured <- t' config
+    let compiled = compile' configured
+    case optimization config of
+      Optimization -> do
+        case go compiled of
```

Comment:
I don't think this is what you want to do. The configuration is for when you want to check whether it's enabled in user code.
Would user code ever do something different depending on whether an optimization flag is set or not? If no, this is the wrong place to set it (as I assume it is).

Reply:
@kozross This is a good point. I don't think we really need optimization configurable since there will rarely be an occasion where code needs to be change depending on it does optimization or not

Reply:
I believe we discussed this, and chose to add this option in case optimizations on IOG's end do something undesirable. I agree that a simpler approach would be to optimize whenever `NoTracing` is requested, and not optimize otherwise; alternatively, _always_ optimize regardless of what tracing is requested. Happy to do either of these.